src/core/job.c

```diff
@@ -187,6 +194,8 @@ static void job_merge_into_installed(Job *j, Job *other) {
 
         j->irreversible = j->irreversible || other->irreversible;
         j->ignore_order = j->ignore_order || other->ignore_order;
+        if (other->triggered_by && j->triggered_by)
```

Comment:
if `other->triggered_by` doesn't exist then it should be created and j->triggered_by added to it, no? Otherwise it would be lost?

Reply:
I think we are merging other into j here, however the check on j->triggered_by should not be there. Pushed a new version of this.